src/net/socket_state.rs

```diff
@@ -78,20 +107,22 @@ impl SocketState {
     }
 
     /// This will process an incoming packet and update acknowledgement information.
-    pub fn process_received(&mut self, addr: SocketAddr, packet: &RawPacket) -> Result<Packet> {
+    pub fn process_received(&mut self, addr: SocketAddr, packet: &PacketHeader) -> Result<()>{
         let connection = self.create_connection_if_not_exists(&addr)?;
         let mut lock = connection
             .write()
-            .map_err(|_| NetworkError::AddConnectionToManagerFailed)?;;
+            .map_err(|_| NetworkError::AddConnectionToManagerFailed)?;
 
         lock.their_acks.ack(packet.seq);
+
         // Update dropped packets if there are any.
-        let dropped_packets = lock.waiting_packets.ack(packet.ack_seq, packet.ack_field);
+        let dropped_packets = lock
+            .waiting_packets
+            .ack(packet.ack_seq, packet.ack_field);
+
         lock.dropped_packets = dropped_packets.into_iter().map(|(_, p)| p).collect();
-        Ok(Packet {
-            addr,
-            payload: packet.payload.clone(),
-        })
+
+        Ok(())
```

Comment:
Why are we not returning a packet here? Im a bit tired so maybe I'm missing something 😄

Reply:
That's right behavior just changed the way this function works.